### PR TITLE
Add Python 3.8 and 3.9 support

### LINKED ISSUE #7: Python 3.8 support
Tests fail on Python 3.8 and 3.9:

- https://travis-ci.com/github/zopefoundation/zope.app.server/jobs/429637788
- https://travis-ci.com/github/zopefoundation/zope.app.server/jobs/429637789

There are two problems:

```
Error in test test_formatter (zope.app.server.tests.test_accesslog.TestAccessLogging)

Traceback (most recent call last):
  File "/opt/python/3.8.3/lib/python3.8/unittest/case.py", line 60, in testPartExecutor
    yield
  File "/opt/python/3.8.3/lib/python3.8/unittest/case.py", line 676, in run
    self._callTestMethod(testMethod)
  File "/opt/python/3.8.3/lib/python3.8/unittest/case.py", line 633, in _callTestMethod
    method()
  File "/home/travis/build/zopefoundation/zope.app.server/src/zope/app/server/tests/test_accesslog.py", line 61, in test_formatter
    logger = self.check_simple_logger("<accesslog>\n"
  File "/home/travis/build/zopefoundation/zope.app.server/src/zope/app/server/tests/test_accesslog.py", line 77, in check_simple_logger
    logger = conf.accesslog()
  File "/home/travis/virtualenv/python3.8.3/lib/python3.8/site-packages/ZConfig/components/logger/factory.py", line 40, in __call__
    self.instance = self.create()
  File "/home/travis/build/zopefoundation/zope.app.server/src/zope/app/server/accesslog.py", line 30, in create
    logger = LoggerFactoryBase.create(self)
  File "/home/travis/virtualenv/python3.8.3/lib/python3.8/site-packages/ZConfig/components/logger/logger.py", line 43, in create
    handler = handler_factory()
  File "/home/travis/virtualenv/python3.8.3/lib/python3.8/site-packages/ZConfig/components/logger/factory.py", line 40, in __call__
    self.instance = self.create()
  File "/home/travis/virtualenv/python3.8.3/lib/python3.8/site-packages/ZConfig/components/logger/handlers.py", line 61, in create
    logger.setFormatter(self.create_formatter())
  File "/home/travis/virtualenv/python3.8.3/lib/python3.8/site-packages/ZConfig/components/logger/formatter.py", line 246, in __call__
    formatter = self.factory(self.format, self.dateformat,
  File "/opt/python/3.8.3/lib/python3.8/logging/__init__.py", line 576, in __init__
    self._style.validate()
  File "/opt/python/3.8.3/lib/python3.8/logging/__init__.py", line 429, in validate
    raise ValueError("Invalid format '%s' for '%s' style" % (self._fmt, self.default_format[0]))
ValueError: Invalid format 'xyzzy' for '%' style
```

and

```
Error in test test_main (zope.app.server.tests.test_main.TestMain)

Traceback (most recent call last):
  File "/opt/python/3.8.3/lib/python3.8/unittest/case.py", line 60, in testPartExecutor
    yield
  File "/opt/python/3.8.3/lib/python3.8/unittest/case.py", line 676, in run
    self._callTestMethod(testMethod)
  File "/opt/python/3.8.3/lib/python3.8/unittest/case.py", line 633, in _callTestMethod
    method()
  File "/home/travis/build/zopefoundation/zope.app.server/src/zope/app/server/tests/test_main.py", line 12, in test_main
    main(['invalid'])
  File "/home/travis/build/zopefoundation/zope.app.server/src/zope/app/server/main.py", line 54, in main
    c0 = time.clock()
AttributeError: module 'time' has no attribute 'clock'
```

### PR DESCRIPTION
**Pull request by mgedmin**
Fixes #7.

### CODE REVIEW

File: src/zope/app/server/tests/test_zpasswd.py [hunk 1]
```diff
@@ -1,7 +1,20 @@
...
+        def assertWarns(self, what):
+            with warnings.catch_warnings(record=True) as w:
+                warnings.simplefilter("always")
+                yield
+                self.assertEqual(len(w), 1)
+                self.assertTrue(issubclass(w[-1].category, what))
+
     def test(self):
-        from zope.app.server.zpasswd import Principal  # noqa
+        with self.assertWarns(DeprecationWarning):
```
Comment (jamadden) [inline src/zope/app/server/tests/test_zpasswd.py:19]: TIL `assertWarns` is a thing.

**Review by jamadden — APPROVED** [review]
LGTM.

File: src/zope/app/server/tests/test_zpasswd.py [hunk 2]
```diff
@@ -1,7 +1,20 @@
 import unittest
+import warnings
+from contextlib import contextmanager
 
 
 class TestZpasswdDeprecation(unittest.TestCase):
 
+    if not hasattr(unittest.TestCase, 'asserWarns'):
```
Comment (icemac) [inline src/zope/app/server/tests/test_zpasswd.py:8]: Sorry for being late to the party but I spotted a typo here:

`asserWarns` -> `assertWarns`

Comment (mgedmin) [reply src/zope/app/server/tests/test_zpasswd.py:8]: oops! :)

thank you

Comment (mgedmin) [reply src/zope/app/server/tests/test_zpasswd.py:8]: Fixed in https://github.com/zopefoundation/zope.app.server/commit/e0734fdc7327a1b41542b664eb745fa4299c2a57

**Pull request merged**